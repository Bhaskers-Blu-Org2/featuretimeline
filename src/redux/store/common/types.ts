export interface IContributionContext {
    level: string;
    team: {
        id: string;
        name: string;
    };
    workItemTypes: string[];
    host: {
        background?: boolean;
    };
}
