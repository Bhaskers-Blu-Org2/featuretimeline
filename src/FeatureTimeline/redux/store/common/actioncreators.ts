import { ActionCreator } from 'redux';
import {
    InitializeAction, InitializeType,
    ResetAction, ResetType,
    TogglePlanFeaturesPaneAction, TogglePlanFeaturesPaneType, ToggleFeatureStateAction, 
    ToggleFeatureStateType, PlanFeaturesPaneWidthChangedAction, PlanFeaturesPaneFilterChangedAction, 
    PlanFeaturesPaneFilterChangedType, PlanFeaturesPaneWidthChangedType
} from './actions';

export const resetAllData: ActionCreator<ResetAction> = () => ({
    type: ResetType,
    payload: null
});

export const createInitialize: ActionCreator<InitializeAction> =
    (projectId: string, teamId: string, backlogLevelName: string) => ({
        type: InitializeType,
        payload: {
            projectId,
            teamId,
            backlogLevelName
        }
    });


export const togglePlanFeaturesPane: ActionCreator<TogglePlanFeaturesPaneAction> = (show: boolean) => ({
    type: TogglePlanFeaturesPaneType,
    payload: show
});

export const changePlanFeaturesWidth: ActionCreator<PlanFeaturesPaneWidthChangedAction> = (width: number) => ({
    type: PlanFeaturesPaneWidthChangedType,
    payload: width
});

export const changePlanFeaturesFilter: ActionCreator<PlanFeaturesPaneFilterChangedAction> = (filter: string) => ({
    type: PlanFeaturesPaneFilterChangedType,
    payload: filter
});


export const toggleFeatureState: ActionCreator<ToggleFeatureStateAction> = (featureName: string, isEnabled: boolean) => ({
    type: ToggleFeatureStateType,
    payload: {
        featureName,
        isEnabled
    }
});
