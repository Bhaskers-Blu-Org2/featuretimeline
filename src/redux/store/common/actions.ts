import { Action } from "redux";

export const InitializeType = "@@common/initialize";
export const ShowDetailsType = "@@common/showdetails";
export const CloseDetailsType = "@@common/closedetails";

export interface InitializeAction extends Action {
    type: "@@common/initialize";
    payload: {
        projectId: string;
        teamId: string;
        backlogLevelName: string;
    }
}

export interface ShowDetailsAction extends Action {
    type: "@@common/showdetails";
    payload: {
        id: number;
    }
}


export interface CloseDetailsAction extends Action {
    type: "@@common/closedetails";
    payload: {
        id: number;
    }
}

export type CommonActions = InitializeAction | ShowDetailsAction | CloseDetailsAction;