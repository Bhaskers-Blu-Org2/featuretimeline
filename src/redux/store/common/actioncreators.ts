import { ActionCreator } from 'redux';
import { InitializeAction, InitializeType, ShowDetailsAction, ShowDetailsType, CloseDetailsAction, CloseDetailsType } from './actions';

export const createInitialize: ActionCreator<InitializeAction> =
    (projectId: string, teamId: string, backlogLevelName: string) => ({
        type: InitializeType,
        payload: {
            projectId,
            teamId,
            backlogLevelName
        }
    });

export const showDetails: ActionCreator<ShowDetailsAction> =
    (id: number) => ({
        type: ShowDetailsType,
        payload: { id }
    });

    
export const closeDetails: ActionCreator<CloseDetailsAction> =
(id: number) => ({
    type: CloseDetailsType,
    payload: { id }
});