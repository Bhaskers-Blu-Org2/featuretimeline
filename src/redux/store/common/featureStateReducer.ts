import { Reducer } from 'redux';
import { ToggleFeatureStateAction, ToggleFeatureStateType } from './actions';
const reducer: Reducer<IDictionaryStringTo<boolean>> = (state: IDictionaryStringTo<boolean> = {}, action: ToggleFeatureStateAction) => {
    switch (action.type) {
        case ToggleFeatureStateType:
            const newState = {...state};
            newState[action.payload.featureName] = action.payload.isEnabled
            return newState;
        default:
            return state;
    }
};

export default reducer;