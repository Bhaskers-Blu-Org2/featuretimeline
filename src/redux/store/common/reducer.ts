import { Reducer } from 'redux';
import { CommonActions, ShowDetailsType, CloseDetailsType } from './actions';
const reducer: Reducer<number[]> = (state: number[] = [], action: CommonActions) => {
    switch (action.type) {
        case ShowDetailsType:
            return [...state, action.payload.id];
        case CloseDetailsType:
            return state.filter(id => id !== action.payload.id);
        default:
            return state;
    }
};

export default reducer;