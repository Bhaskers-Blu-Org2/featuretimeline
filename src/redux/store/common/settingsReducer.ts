import { Reducer } from 'redux';
import { ISettingsState, ProgressTrackingCriteria } from '../types';
import { SettingsActions, ToggleShowWorkitemDetailsType, ChangeProgressTrackingCriteriaType, RestoreSettingsType, ChangeShowClosedSinceDaysType } from './actions';
import produce from "immer";

export const getDefaultSettingsState = (): ISettingsState => {
    return {
        showWorkItemDetails: false,
        progressTrackingCriteria: 0,
        showClosedSinceDays: 0
    };
}
const reducer: Reducer<ISettingsState> = (state: ISettingsState = getDefaultSettingsState(), action: SettingsActions) => {
    const {
        type,
        payload
    } = action;
    return produce(state, draft => {
        switch (type) {
            case ToggleShowWorkitemDetailsType:
                draft.showWorkItemDetails = payload as boolean;
            case ChangeProgressTrackingCriteriaType:
                draft.progressTrackingCriteria = payload as ProgressTrackingCriteria;
            case ChangeShowClosedSinceDaysType:
                draft.showClosedSinceDays = payload as number;
            case RestoreSettingsType:
                return payload as ISettingsState;
        }
    });
};

export default reducer;