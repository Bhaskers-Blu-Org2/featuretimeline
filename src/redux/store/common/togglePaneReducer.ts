import { Reducer } from 'redux';
import { TogglePlanFeaturesPaneType, PlanFeaturesPaneFilterChangedType, PlanFeaturesPaneActions, PlanFeaturesPaneWidthChangedType } from './actions';
import { IPlanFeaturesState } from '../types';
import produce from "immer";

export const getDefaultPlanFeaturesPaneState = (): IPlanFeaturesState => {
    return {
        show: false,
        paneWidth: 300,
        filter: null
    };
}
const reducer: Reducer<IPlanFeaturesState> = (state: IPlanFeaturesState = getDefaultPlanFeaturesPaneState(), action: PlanFeaturesPaneActions) => {
    const {
        type,
        payload
    } = action;

    return produce(state, draft => {
        switch (type) {
            case TogglePlanFeaturesPaneType:
                draft.show = payload as boolean;
                break;
            case PlanFeaturesPaneFilterChangedType:
                draft.filter = payload as string;
                break;
            case PlanFeaturesPaneWidthChangedType:
                draft.paneWidth = payload as number;
                break;
        }
    });
};

export default reducer;