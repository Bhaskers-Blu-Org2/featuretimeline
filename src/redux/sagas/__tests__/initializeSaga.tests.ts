declare var it, expect;

import { callinitialize } from '../initialize';
import { createInitialize } from '../../store/common/actioncreators';

it('test initialize saga basic test', () => {
    const initializeAction = createInitialize("projectId", "teamId", "backlogLevel");
    const saga = callinitialize(initializeAction);

    // Expect loading true
    expect(saga.next()).toMatchSnapshot();

    // Expect call to initialize saga
    expect(saga.next()).toMatchSnapshot();

    // Expect loading false
    expect(saga.next()).toMatchSnapshot();

    // Expect null
    expect(saga.next()).toMatchSnapshot();
});
