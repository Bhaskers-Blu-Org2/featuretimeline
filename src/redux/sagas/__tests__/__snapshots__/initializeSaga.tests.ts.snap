// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test initialize saga basic test 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "PUT": Object {
      "action": Object {
        "payload": true,
        "type": "@@loading/loading",
      },
      "channel": null,
    },
  },
}
`;

exports[`test initialize saga basic test 2`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "CALL": Object {
      "args": Array [
        Object {
          "payload": Object {
            "backlogLevelName": "backlogLevel",
            "projectId": "projectId",
            "teamId": "teamId",
          },
          "type": "@@common/initialize",
        },
      ],
      "context": null,
      "fn": [Function],
    },
  },
}
`;

exports[`test initialize saga basic test 3`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "PUT": Object {
      "action": Object {
        "payload": false,
        "type": "@@loading/loading",
      },
      "channel": null,
    },
  },
}
`;

exports[`test initialize saga basic test 4`] = `
Object {
  "done": true,
  "value": undefined,
}
`;
